### ISSUE TITLE
Use S3D from training multi-class binary for video classification??

### ISSUE THREAD

**Issue by lukens94** [kylemin/S3D#8]
I should train the model to recognize a series of (volcanic) videos which can belong to 8 different classes (all or only some of them) For example:

-video 1 belongs to classes 1,4,6
-video 2 belongs to class 3
-video 3 belongs to classes 2,5
etc...
Each class has a binary label 1 (belongs) or 0 (does not belong).

I thought of calculating the loss for each of these 8 classes by specifying in each of these the weight of that specific class.
This is my class_weight = [5.56, 51.5, 3.67, 1.39, 5.36, 2.82, 5.18, 1.76]
![image](https://user-images.githubusercontent.com/44710933/125939708-e5fb279c-3b22-4a80-89fa-8ec0ee856eaa.png)

My dataset is structured as follows:
210 videos of 300 frames each, of these I took one every five frames and combined in 29 different combinations each of 32 frames:
frames 1,2,3,4,5,6 .... 300 -> frames 1,6,11,16,21 ..... (total 60 frames):
first combination: first 32 frames (1-32)
second combination: second 32 frames shifted by 1 (2-33)
third combination: third 32 frames shifted by 1 (3-34)
etc....
total array 210x29x32

The input of the s3d model are these 32 frames concatenated with pytorch and resized: 
**torch.Size ([2, 3, 32, 224, 224])**
associated with a list of 8 targets each relating to a class (1 or 0) 

![image](https://user-images.githubusercontent.com/44710933/125942947-c4b571ae-a507-4043-8ee1-c53655a17906.png)

This is my fit function where I am going to pass my targets and my model output inside binary_loss_fn. I specified the number of classes when I initialize the model
**model = S3D (8) .to (device)**
but i can't understand what kind of output i get on these s3d because i can't calculate individually my 8 loss functions.

This is a type output with batch equal to 2:
torch.tensor ([
[0.4150, 0.6833, 0.4165, 0.4718, 0.4183, 0.5799, 0.4773, 0.3026],
[0.4923, 0.5274, 0.4894, 0.5042, 0.4440, 0.4746, 0.4580, 0.4449]
])

How could I fix this. Thank you
